### DISCUSSION TITLE
Image resizing

### DISCUSSION THREAD

**Discussion by rafaelzemog** [Witiko/markdown#280] [Q&A] [answered]
As discused in [#49](https://github.com/Witiko/markdown/issues/49) about the image example.

I have a lot of images in my markdown files but when I try to resize them the parser does nothing. Reading the documentation the parser only takes the name of the file and adjust the size to \textwidth. Is there any adittional option to include in order to resize any included image in markdown?
As far as I know an image can be resized with:
```
![example image](example-image.png "An example image from Martin Scharrer's mwe package" =250x250)
```

**Comment by Witiko** [comment]
As you discovered, there is currently no support for image resizing, although you can [redefine the image renderer](http://mirrors.ctan.org/macros/generic/markdown/markdown.html#image-renderer) to extract additional information from the label, such as:

``` markdown
![250x250](example-image.png "An example image from Martin Scharrer's mwe package")
```

Note that TeX does not deal in pixels, so the size only makes sense if you assume some depth per inch (such as 72 DPI, which is used on the web), and use this assumption to convert to units that TeX understands (250px at 72 DPI would be 3.47 inches or 8.81 cm, both of which are understood by TeX).

There are many markdown syntax extensions that this package does not support, but I have not come across this one. Out of curiosity, which tool are you using that can resize images using this syntax?

**Comment by rafaelzemog** [comment]
My environment is Atom using the Markdown Preview package I also write in MacDown. My main OS is Mac OSX Mojave. Basically I just  moved from MS Office and made my own latex template fed by markdown files. After a few online research I found this instructions in order to adjust the maximum size of any imported image.
```
\makeatletter
\setkeys{Gin}{width=\ifdim\Gin@nat@width>\linewidth
    0.5\linewidth
\else
    \Gin@nat@width
\fi}
\makeatother
```
Thanks for your fast reply, as far as I am understanding the markdown package I will write my own renderers in order to enhance this awesome package and I will post it.

**Comment by Witiko** [comment]
Thank you for sharing your setup and your solution. The aim of the Markdown package is to allow the author to leverage both their knowledge of Markdown and their knowledge of (La)TeX. I am looking forward to seeing your renderers.

As for the syntax extension, the Atom's Markdown Preview package [seems](https://github.com/atom/markdown-preview/blob/master/package.json#L16) to be using the [Marked](https://marked.js.org/) markdown parser. The documentation of Marked only claims the support of two Markdown flavors: [CommonMark](http://spec.commonmark.org/0.28/) and [GitHub Flavored Markdown](https://github.github.com/gfm/). To my knowledge, neither flavor supports the above syntax for resizing images. However, if we manage to track down the origin (and the documentation) of this syntax extension, I am in favor of adding it to the Markdown package.

**Comment by Witiko** [comment]
Any news on this, @rafaelzemog?

**Comment by Migacz85** [comment]
I'm compiling with pdflatex and include markdown like that:

`\usepackage[fencedCode,taskLists, pipeTables, tableCaptions, tightLists, citations, hybrid]{markdown}`

I've tried it like that:

![image](https://user-images.githubusercontent.com/25789997/208254280-4e6124b7-35ae-4d1a-a7b6-448bd533a6e4.png)

But somehow text from markdown don't want to comply:

![image](https://user-images.githubusercontent.com/25789997/208254286-f45739fd-a4e9-4ec8-b3fd-48393ee4d765.png)



But without markdown blocks it will work. Any ideas?

![image](https://user-images.githubusercontent.com/25789997/208254490-1ddf3a72-e34a-489e-a689-b28be5362981.png)

**Comment by Witiko** [comment]
@Migacz85 Not sure what kind of trick wrapfig uses that markdown breaks. Does it help to remove the `\end{markdown}` and `\begin{markdown}` around the wrapfig, or include the markdown text from an external file using `\markdownInput{asdf.md}`?

**Comment by Migacz85** [comment]
I should be more specific the last figure with correctly align picture is without tags `begin{markdown}, \end{markdown}` .  End when you will add text with `\markdownInput` - same bug.

**Comment by Witiko** [comment]
> I should be more specific the last figure with correctly align picture is without tags begin{markdown}, \\end{markdown} .

Here is what I meant:

![208254280-4e6124b7-35ae-4d1a-a7b6-448bd533a6e4.png](https://user-images.githubusercontent.com/603082/208265380-9dc08008-ec89-44c1-a2c3-4edf963db6dd.png)

Enabling the `hybrid` option should allow you to type LaTeX without dropping out of the markdown environment.

**Comment by Migacz85** [comment]
I just tested this, and unfortunately still doesn't work. I'm using hybrid
and when I pasted code for wrapping directly in Markdown the figure was on
the right, and the text was going on the figure.

**Comment by Witiko** [comment]
[The `wrapfig` documentations][1] says that you can specify the number of lines that will be pushed aside (here 20) using an optional parameter:

``` tex
\begin{wrapfigure}[20]{r}{0.4\textwidth}
```

I am not sure how `wrapfig` does its pushing-aside, but this may help.

 [1]: http://mirrors.ctan.org/macros/latex/contrib/wrapfig/wrapfig-doc.pdf

**Comment by Witiko** [comment, marked as answer]
Since #256 (to be released in version 2.22.0 of the Markdown package at the end of March), you can use [the `linkAttribute` option][1] to specify attributes for links and images.

``` tex
\documentclass{article}
\usepackage[linkAttributes]{markdown}
\begin{document}
\begin{markdown}
![example image](example-image.png "An example image from Martin Scharrer's mwe package"){width=250mm height=250mm}
\end{markdown}
\end{document}
```

In the Markdown package for LaTeX, key-value attributes on images are passed to the `\includegraphics` command by default, i.e. the above translates roughly to `\includegraphics[width=250mm, height=250mm]{example-image.png}`.

 [1]: https://witiko.github.io/markdown/#option-linkattributes
